### PR TITLE
Release Notes SAM 2024.12.12

### PR DESCRIPTION
**Pull request by cpaulgilman**
Release notes for Fall 2024 release.

To view as html: https://htmlpreview.github.io/?https://github.com/NREL/SAM/blob/gh-pages-2024/doc/releasenotes.html

To edit in web browser: https://github.com/NREL/SAM/blob/gh-pages-2024/doc/releasenotes.html

** *Do not merge into SAM/gh-pages until SAM and SSC version numbers are updated and installation files are posted on SAM website Downloads page* **

This gh-pages branch is a special documentation branch that does not contain any code. Release notes are automatically published at https://nrel.github.io/SAM/doc/releasenotes.html. SAM website redirect points to that URL: https://sam.nrel.gov/releasenotes.

### CODE REVIEW

File: doc/releasenotes.html [hunk 1]
```diff
@@ -48,17 +54,205 @@ <h1 id="yyyy-mm-dd-r[n]">SAM yyyy.m.d [r[n]], SSC nnn: Month d, yyyy</h1>
...
+
+<h2>Detailed PV (SAM 2024.12.12)</h2>
+
+<p>Improve solar position algorithm to speed up PVWatts and Detailed PV simulation run times. (<a href="https://github.com/NREL/ssc/pull/1212">SSC 1212</a>)</p>
+
+<p>Update module and inverter libraries. (<a href="https://github.com/NREL/SAM/pull/1913">SAM 1913</a>, <a href="https://github.com/NREL/SAM/pull/1917">SAM 1917</a>)</p>
+
+<p>Rename temperature coefficient of power variable from 'gamma_r' to 'gamma_pmp' to be more consistent with industry standard. (<a href="https://github.com/NREL/ssc/pull/1231">SSC 1231</a>, <a href="https://github.com/NREL/SAM/pull/1898">SAM 1898</a>)</p>
+
+<p>Add subhourly clipping correction option based on matrix lookup approach ("Allen Method"). (<a href="https://github.com/NREL/SAM/pull/1861">SAM 1861</a>)</p>
```
Comment (brtietz) [inline doc/releasenotes.html:162]: @mjprilliman Can you take a look at this? I'm tempted to change it to "add to the GUI" but that doesn't seem quite right here.

File: doc/releasenotes.html [hunk 2]
```diff
@@ -48,17 +54,205 @@ <h1 id="yyyy-mm-dd-r[n]">SAM yyyy.m.d [r[n]], SSC nnn: Month d, yyyy</h1>
...
+
+<p>Improve solar position algorithm to speed up PVWatts and Detailed PV simulation run times. (<a href="https://github.com/NREL/ssc/pull/1212">SSC 1212</a>)</p>
+
+<h2>Electric Battery Storage (SAM 2024.12.12)</h2>
+
+<p>Update and fix REopt API calls for battery size and dispatch optimization. (<a href="https://github.com/NREL/SAM/pull/1904">SAM 1904</a>, <a href="https://github.com/NREL/ssc/pull/1242">SSC 1242</a>, <a href="https://github.com/NREL/ssc/pull/1203">SSC 1203</a>)</p>
+
+<p>Add battery-specific availability loss inputs. (<a href="https://github.com/NREL/SAM/pull/1902">SAM 1902</a>, <a href="https://github.com/NREL/ssc/pull/1241">SSC 1241</a>, <a href="https://github.com/NREL/ssc/pull/1234">SSC 1234</a>)</p>
+
+<p>Modify battery dispatch to consider grid limits. (<a href="https://github.com/NREL/SAM/pull/1883">SAM 1883</a>, <a href="https://github.com/NREL/ssc/pull/1224">SSC 1224</a>)</p>
```
Comment (brtietz) [inline doc/releasenotes.html:186]: Having this third feels a little like burying the lede on the battery changes. Mind putting it first?

**Review by brtietz — CHANGES_REQUESTED** [review]
Noting a few typos on a quick review before my first meeting. I'll come back for more around 10 am.

File: doc/releasenotes.html (hunk 1 above)
Comment (mjprilliman) [reply doc/releasenotes.html:162]: I would describe it as the following: 

'Add PV output distribution clipping correction approach ("Walker method"). Update the user interface to allow the user to select either the Allen or Walker method for hourly simulations

File: doc/releasenotes.html [hunk 3]
```diff
@@ -48,17 +54,205 @@ <h1 id="yyyy-mm-dd-r[n]">SAM yyyy.m.d [r[n]], SSC nnn: Month d, yyyy</h1>
...
+
+<p>Update module and inverter libraries. (<a href="https://github.com/NREL/SAM/pull/1913">SAM 1913</a>, <a href="https://github.com/NREL/SAM/pull/1917">SAM 1917</a>)</p>
+
+<p>Rename temperature coefficient of power variable from 'gamma_r' to 'gamma_pmp' to be more consistent with industry standard. (<a href="https://github.com/NREL/ssc/pull/1231">SSC 1231</a>, <a href="https://github.com/NREL/SAM/pull/1898">SAM 1898</a>)</p>
+
+<p>Add subhourly clipping correction option based on matrix lookup approach ("Allen Method"). (<a href="https://github.com/NREL/SAM/pull/1861">SAM 1861</a>)</p>
+
+<p>Fix six-parameter module coefficient generator sanity check that caused a crash. This should result in more modules from the CEC list to be inlcuded in the SAM module library. (<a href="https://github.com/NREL/ssc/pull/1245">SSC 1245</a>)</p>
+
+<p>Fix solar position algorithm issue that caused SAM to crash when running multi-threaded such as for parametric simulations. (<a href="https://github.com/NREL/ssc/pull/1232">SSC 1232</a>)</p>
```
Comment (mjprilliman) [inline doc/releasenotes.html:166]: I believe this is a dev bug, as the issue with multi-threading only happened once the hash table fix was merged in.

File: doc/releasenotes.html [hunk 4]
```diff
@@ -48,17 +54,205 @@ <h1 id="yyyy-mm-dd-r[n]">SAM yyyy.m.d [r[n]], SSC nnn: Month d, yyyy</h1>
...
+
+<p>Add battery-specific availability loss inputs. (<a href="https://github.com/NREL/SAM/pull/1902">SAM 1902</a>, <a href="https://github.com/NREL/ssc/pull/1241">SSC 1241</a>, <a href="https://github.com/NREL/ssc/pull/1234">SSC 1234</a>)</p>
+
+<p>Modify battery dispatch to consider grid limits. (<a href="https://github.com/NREL/SAM/pull/1883">SAM 1883</a>, <a href="https://github.com/NREL/ssc/pull/1224">SSC 1224</a>)</p>
+
+<p>Add new Standalone Battery input to define simulation time step rather than basing it on other inputs. (<a href="https://github.com/NREL/SAM/pull/1887">SAM 1887</a>, <a href="https://github.com/NREL/ssc/pull/1227">SSC 1227</a>)</p>
+
+<p>Add metrics and graphs to Summary tab on Results page, and default graph showing battery performance metrics on Time Series tab. (<a href="https://github.com/NREL/SAM/pull/1884">SAM 1884</a>, <a href="https://github.com/NREL/SAM/pull/1865">SAM 1865</a>, <a href="https://github.com/NREL/ssc/pull/1215">SSC 1215</a>)</p>
```
Comment (mjprilliman) [inline doc/releasenotes.html:190]: May want to qualify this as BTM battery metrics, could be convinced either way

File: doc/releasenotes.html [hunk 5]
```diff
@@ -48,17 +54,205 @@ <h1 id="yyyy-mm-dd-r[n]">SAM yyyy.m.d [r[n]], SSC nnn: Month d, yyyy</h1>
...
+<li>PVWatts default system designs assume monofacial modules because bifacial option is hidden in collapsed "Advanced Parameters" panel. See <a href="https://github.com/NREL/SAM/issues/1959">SAM Issue 1959</a>.</li>
+</ul>
+
+<p>For IPH configurations, update default incentives (no investment tax credit, ITC), heat sink capacity, and O&M cost. (<a href="https://github.com/NREL/SAM/pull/1946">SAM 1946</a>)</p>
+
+<p>Update default utility rate for behind-the-meter (BTM) configurations. (<a href="https://github.com/NREL/SAM/pull/1958">SAM 1958</a>)</p>
+
+<h2>Financial Models (SAM 2024.12.12)</h2>
+
+<p>Update investment-tax credit (ITC) and depreciation basis calculations to remove financing fees and reserve account funding from basis. <a href="https://github.com/NREL/SAM/pull/1860">SAM 1860</a>, <a href="https://github.com/NREL/ssc/pull/1193">SSC 1193</a>)</p>
```
Comment (brtietz) [inline doc/releasenotes.html:139]: Missing an open parenthesis before the GH PR links

File: doc/releasenotes.html [hunk 6]
```diff
@@ -48,17 +54,205 @@ <h1 id="yyyy-mm-dd-r[n]">SAM yyyy.m.d [r[n]], SSC nnn: Month d, yyyy</h1>
...
+
+<p>Improve efficiency of internal handling of battery parameter variables to decrease simulation run time, particularly for PV smoothing dispatch option. (<a href="https://github.com/NREL/ssc/pull/1251">SSC 1251</a>)</p>
+
+<p>Change battery life model to base degraded battery capacity on sum of calendar and cycle degradation rather than the minimum of the two. (<a href="https://github.com/NREL/ssc/pull/1239">SSC 1239</a>)</p>
+
+<p>Improve determination of price signal forecast for Merchant Plant configurations with automated dispatch: Calculate price signal in each time step as capacity-weighted sum of enabled energy and ancillary service prices instead of simple sum. (<a href="https://github.com/NREL/ssc/pull/1235">SSC 1235</a>)</p> 
+
+<h2>Hybrid Systems (SAM 2024.12.12)</h2>
+
+<p>Add input pages for thermal loads and thermal rates to hybrid configurations that include fuel cells. <a href="https://github.com/NREL/SAM/pull/1903">SAM 1903</a>, <a href="https://github.com/NREL/ssc/pull/1244">SSC 1244</a>)</p>
```
Comment (brtietz) [inline doc/releasenotes.html:200]: Missing open parenthesis before PR links

File: doc/releasenotes.html [hunk 7]
```diff
@@ -48,17 +54,205 @@ <h1 id="yyyy-mm-dd-r[n]">SAM yyyy.m.d [r[n]], SSC nnn: Month d, yyyy</h1>
...
+
+<p>Include battery discharge power in power profile used to calculate project costs and benefits. (<a href="https://github.com/NREL/ssc/issues/1256">SSC 1256</a>)</p>
+
+<h2>Concentrating Solar Power (CSP) (SAM 2024.12.12)</h2>
+
+<p>For parabolic trough and linear Fresnel configurations, move input variables for receiver startup dely time and energy fraction from Solar Field page to System Control page. These inputs are for the dispatch optimization algorithm, not the performance simulation. (They are inputs to the performance model for power tower configurations.) (<a href=<https://github.com/NREL/SAM/pull/1961>SAM 1961</a>)/p>
```
Comment (brtietz) [inline doc/releasenotes.html:206]: Stray html at end of line `/p>`

**Comment by janinefreeman** [comment]
I'd love to pull the new BTM PV+battery+utility rate+load visualizations up into "highlights", as well as the new compare cases macro. If that's too many items, I'd jettison module/inverter library updates. We may also want to mention the Walker algorithm in highlights, thoughts from @mjprilliman ? No other additional comments from me!

File: doc/releasenotes.html (hunk 2 above)
Comment (cpaulgilman) [reply doc/releasenotes.html:186]: @brtietz I didn't consider order -- I'd welcome any other suggestions for putting things in order of importance.

**Review by brtietz — APPROVED** [review]
Thanks for updating!

**Pull request merged**